src/celeritas/field/CartMapFieldInput.hh

```diff
@@ -0,0 +1,70 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/field/CartMapFieldInput.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <vector>
+
+#include "corecel/Types.hh"
+#include "celeritas/Types.hh"
+
+#include "FieldDriverOptions.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Input data for a magnetic X-Y-Z vector field stored on an X-Y-Z grid.
+ *
+ * The magnetic field is discretized at nodes on an X-Y-Z grid, and at each
+ * point the field vector is approximated by a 3-D vector in X-Y-Z. The input
+ * units of this field are in *NATIVE UNITS* (cm/gauss when CGS). An optional
+ * \c _units field in the input can specify whether the input is in SI or CGS
+ * units, with allowable values of "si", "cgs", or "clhep". The native CLHEP
+ * unit strength is 1000*tesla.
```

Comment:
```suggestion
 * point the field vector is approximated by a 3-D vector in X-Y-Z. The input
 * units of this field are in *NATIVE CELERITAS UNITS* (cm/gauss when CGS).
```
since this doesn't have JSON IO